engine/crates/common-types/src/lib.rs

```diff
@@ -36,6 +45,13 @@ pub enum LogLevel {
     Debug,
 }
 
+#[serde_with::serde_as]
```

Comment:
nitpicking, but doesn't look like we need serde_as here?

Reply:
Good catch!